File: lc/T151-200/t167.cc

#include <bits/stdc++.h>
using namespace std;
class Solution
{
public:
    vector<int> twoSum(vector<int> &numbers, int target)
    {
        map<int, int> m;
        for (int i = 0; i < numbers.size(); i++)
        {
            if (m.count(numbers[i]))
            {
                return {i+1, m[numbers[i]]};
            }
            m[target - numbers[i]] = i+1;
        }
        return {-1, -1};
    }
};

int main()
{
    vector<int> v{2, 7, 11, 15};
    Solution s;
    auto it = s.twoSum(v, 9);
    for (auto i : it)
        cout << i << " ";
    cout << endl;


    int n  =1 ; 
    cout<< ~n<<endl;
    return 0;
}

File: leetcode/lc/T151-200/t_167.java
package leetcoder;

import java.util.Arrays;
import java.util.HashMap;
import java.util.Map;


class Solution {
    public int[] twoSum(int[] numbers, int target) {
        int [] res = new int[2];
        for (int i = 0; i < numbers.length; i++) {
            for (int j = i+1; j < numbers.length; j++) {
                if (numbers[i] + numbers[j] == target) {
                    res[0] = i+1;
                    res[1] = j+1;
                    return res;
                }
            }
        }
        return res;
    }
}



public class Main {

    public static void main(String[] arg){
    }

}